### ISSUE TITLE
Upscales images and generates files larger than original

### ISSUE THREAD

**Issue by paleite** [Levelmoney/resize-image-loader#15]
The plugin generates versions that are larger than the original, which results in larger images and larger files than the original.

In my opinion, rather than returning an image that is larger than the original, the original image should be used.
In my example below, the output should ideally be an empty srcset, seeing as the images with smaller image size have a larger filesize than the original, the equal image has a larger filesize and larger images shouldn't exist to begin with.

Example: Using a [test-image](https://upload.wikimedia.org/wikipedia/commons/c/c4/PM5544_with_non-PAL_signals.png)
src/test-image.png (dimensions: 768x576) => 15118 bytes

dist/test-image-1200w.png => 92303 bytes (larger image size, larger filesize compared to original)
dist/test-image-992w.png => 68055 bytes  (larger image size, larger filesize compared to original)
dist/test-image-768w.png => 15760 bytes  (equal image size, larger filesize compared to original)
dist/test-image-480w.png => 34512 bytes  (smaller image size, larger filesize compared to original, larger filesize compared to 768w version)

I understand if it's a bit hard to understand what I mean, but I hope it makes sense.

TL;DR:
- No image sizes larger than the original should be generated
- If the resulting image has the same dimensions as the original image but a larger filesize, no image should be generated
- No images smaller than the original should be larger in filesize than the original
- No image size smaller than a previous image size should be larger in filesize (i.e. 480w version MUST be smaller in filesize than 768w version, otherwise 768w should be used)

Hope that makes sense 👍

**Comment by webyak** [comment]
Any updates on this?

**Comment by puppybits** [comment]
To not upscale images that's a simple fix (unless they set an override flag). 

For the smaller images having a larger file size, I can look at the settings of the image compression to be more aggressive. I haven't had the time recently and I've since left LevelMoney so I'm trying to get access to approve my own pull requests. I need that cleared up (or fork this) before I do too much.
